modules/API/API.js

```diff
@@ -236,7 +237,11 @@ function initCommands() {
         'pin-participant': id => {
             logger.debug('Pin participant command received');
             sendAnalytics(createApiEvent('participant.pinned'));
-            APP.store.dispatch(pinParticipant(id));
+            if (isStageFilmstripAvailable(APP.store.getState())) {
+                APP.store.dispatch(addStageParticipant(id, true));
+            } else {
+                APP.store.dispatch(pinParticipant(id));
```

Comment:
Note for discussion at some point. I'd say the `pinParticipant` is a natural place to encapsulate this logic...